test/hack/cleanup/main.go

```diff
@@ -441,6 +443,71 @@ func (ip *instanceProfile) Cleanup(ctx context.Context, names []string) ([]strin
 	return deleted, errs
 }
 
+type eni struct {
+	ec2Client *ec2.Client
+}
+
+func (e *eni) Type() string {
+	return "ElasticNetworkInterface"
+}
+
+func (e *eni) Get(ctx context.Context, expirationTime time.Time) (ids []string, err error) {
+	var nextToken *string
+	for {
+		out, err := e.ec2Client.DescribeNetworkInterfaces(ctx, &ec2.DescribeNetworkInterfacesInput{
```

Comment:
this manual pagination works fine, but you could also use the built-in paginator: https://aws.github.io/aws-sdk-go-v2/docs/making-requests/#using-paginators

Reply:
I will update the paginator for all the resources in a separate PR